src/Ocelot/Middleware/Pipeline/OcelotPipelineExtensions.cs

```diff
@@ -36,6 +36,12 @@ public static class OcelotPipelineExtensions
                     app.UseDownstreamRouteFinderMiddleware();
                     app.UseDownstreamRequestInitialiser();
                     app.UseLoadBalancingMiddleware();
+
```

Comment:
This should be right after the DownstreamRouteFinderMiddleware() call if you are going to use it.

Reply:
@Rakiah please can you take a look into @TomPallister  comment?